bibliopixel/animation/animation.py

```diff
@@ -23,21 +33,44 @@ def _led(self, layout):
     def preRun(self, amt=1):
         self.layout.all_off()
 
+    @property
+    def complete_callback(self):
+        return self.__complete_callback
+
+    @complete_callback.setter
+    def complete_callback(self, callback):
+        if not callable(callback):
+            raise ValueError('complete_callback must be a function!')
+
+        self.__complete_callback = callback
+
     def step(self, amt=1):
         raise RuntimeError("Base class step() called. This shouldn't happen")
 
-    def cleanup(self):
-        self.threading.stop_thread(wait=True)
-        self.layout.cleanup()
+    def cleanup(self, clean_layout=True):
+        # if current thread is animation thread this was called
+        # by the context manager and thread is therefore already stopped
+        if self.threading.thread != threading.current_thread():
+            self.threading.stop_thread(wait=True)
```

Comment:
That's legacy and was already in there. Probably not an issue if no wait... will play around.